src/rgw/driver/sfs/object.cc

```diff
@@ -282,26 +279,91 @@ void SFSObject::gen_rand_obj_instance_name() {
   state.obj.key.set_instance(buf);
 }
 
-int SFSObject::get_obj_attrs(
-  optional_yield y,
-  const DoutPrefixProvider *dpp,
-  rgw_obj *target_obj
-) {
-  lsfs_dout(dpp, 10) << " TODO" << dendl;
+/*
+This must be intended as a fetch of object's attributes.
+Caller then will likely call get_attrs().
+*/
+int SFSObject::get_obj_attrs(optional_yield y,
+                             const DoutPrefixProvider *dpp,
+                             rgw_obj *target_obj) {
+  refresh_meta();
```

Comment:
Is there no expectation that the object be returned in `target_obj`? It seems weird that we have a `get` operation that is just refreshing metadata.

Reply:
This is how I have interpreted this call.
I have checked around and the usage seems to suggest that one first calls `get_obj_attrs` and then it uses the `get_attrs`.
I have serious doubts that this guess is entirely correct for all cases; so for the time being I would leave `target_obj` empty and rely on the fact that in the future something could potentially fail because of this. But this failing will have a more precise semantics that a potentially fuzzy failing because now we are filling `target_obj` with what we think is our current guess.

Reply:
if that's the case, I'd suggest adding a comment about that in the function, and assert in case your expectations are not met. That way we can potentially catch corner cases.